File: output/opt3001-driver/examples/python_example/opt3001_demo.py
#!/usr/bin/env python3

import sys
import os
sys.path.append(os.path.join(os.path.dirname(__file__), '..', 'src', 'python'))

from opt3001 import OPT3001

def main():
    try:
        sensor = OPT3001(bus_number=1)
        lux = sensor.read_lux()
        print(f"Light level: {lux:.2f} lux")
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    main()

File: output/opt3001-driver/examples/ts_example/opt3001_demo.ts
import { openPromisified } from 'i2c-bus';
import { Opt3001 } from '../src/ts/opt3001';

async function main() {
    try {
        const bus = await openPromisified(1);
        const sensor = new Opt3001(bus);
        const lux = await sensor.readLux();
        console.log(`Light level: ${lux.toFixed(2)} lux`);
        await bus.close();
    } catch (error) {
        console.error('Error:', error);
    }
}

main();
